ansible/roles/csm.cn.dns-resolver/tasks/main.yaml

```diff
@@ -0,0 +1,64 @@
+# Copyright 2023 Hewlett Packard Enterprise Development LP
+# resolver config tasks
+---
+- block:
+  - name: Slurp in /etc/resolv.conf to get any DHCP added entries
+    slurp:
+      src: "/etc/resolv.conf"
+    register: resolv_conf
+
+  - name: Register /etc/resolv.conf contents
+    set_fact:
+      resolv: "{{ resolv_conf.content | b64decode }}"
+
+  - name: Parse existing options from /etc/resolv.conf
+    set_fact:
+      node_existing_options: "{{ resolv | regex_findall( 'options (\\b(?:[0-9]{1,3}\\.){3}[0-9]{1,3})\\b') }}"
+
+  - name: Slurp in /etc/sysconfig/network/config to get any DHCP added entries
+    slurp:
+      src: "/etc/sysconfig/network/config"
+    register: net_config
+
+  - name: Register /etc/sysconfig/network/config contents
+    set_fact:
+      netconf: "{{ net_config.content | b64decode }}"
+
+  - name: Parse options from config list
+    set_fact:
+      node_netconf_options: "{{ item.split() }}"
+    loop: "{{ netconf | default() | regex_findall( 'NETCONFIG_DNS_RESOLVER_OPTIONS=\"(.*)\"') }}"
+
+  - name: Add existing options to node_netconf_options
+    set_fact:
+      node_netconf_options: "{{ node_netconf_options + [ item ] }}"
+    when: (node_existing_options is defined) and (item not in node_netconf_options)
+    loop: "{{ node_existing_options }}"
+
+  - name: Add required options to node_netconf_options
+    set_fact:
+      node_netconf_options: "{{ node_netconf_options + [ item ] }}"
+    when: item not in node_netconf_options
+    loop: "{{ node_required_dns_options }}"
+
+  - name: Add any customer defined options to node_netconf_options
+    set_fact:
+      node_netconf_options: "{{ node_netconf_options + [ item ] }}"
+    when: item not in node_netconf_options
+    loop: "{{ node_external_dns_options }}"
+
+  - name: Set DNS_RESOLVER_OPTIONS entries
+    when: node_netconf_options is defined
+    lineinfile:
+      dest: /etc/sysconfig/network/config
+      regex: '^NETCONFIG_DNS_RESOLVER_OPTIONS=\"(.*)\"$'
+      line: "NETCONFIG_DNS_RESOLVER_OPTIONS=\"{{ node_netconf_options | join(' ') }}\""
+      backrefs: yes
+
+  - name: Regenerate /etc/resolv.conf
+    shell: netconfig update -f
+    changed_when: result.rc == 0
+    failed_when: result.rc != 0
+    register: result
+
+  when: not cray_cfs_image
```

Comment:
This is not the recommended way or limiting to image customization / node personalization any more.  While it looks clean to place a single "when" statement here, Ansible expands this at run-time and not only evaluates it for every task, it's going to evaluate it for every loop iteration of every task.  It's not efficient, and can add up to several seconds of wasted time in the long run.  (Yes I'm being nit-picky, but also yes we've had at least one customer complain about the time wasted evaluating "when" statements")  See https://github.com/Cray-HPE/docs-csm/blob/main/operations/configuration_management/Target_Ansible_Tasks_for_Image_Customization.md for the new method.